### ISSUE TITLE
Small Bug in Encoder Implementation

### ISSUE THREAD

**Issue by manoja328** [huggingface/transformers#28902]
### System Info

any version , latest version

### Who can help?

@ArthurZucker   @younesbelkada

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

https://github.com/huggingface/transformers/blob/a1afec9e1759b0fdb256d41d429161cc15ecf500/src/transformers/models/mobilebert/modeling_mobilebert.py#L585 

head_mask[i] will not exists if head_mask supplied is None. And this will give an error. 

Solution: in the inner loop  
`head_mask = head_mask[i] if head_mask is not None else None`

### Expected behavior

Solution: in the inner loop  
`head_mask = head_mask[i] if head_mask is not None else None`

**Comment by manoja328** [comment]
Got and error when tried to pass embeddings 

output = encoder(embeddings)

**Comment by younesbelkada** [comment]
Hi @manoja328 
Thanks for the issue! This is I think expected as this class is not meant to be used as a standalone class, `head_mask` is automatically created as a list of `None` here: https://github.com/huggingface/transformers/blob/1c31b7aa3bb4e7ef24c77596d2a76f45a770159f/src/transformers/modeling_tf_utils.py#L1219
but I agree we could make it more userfriendly, one way could be to initialize `head_mask` in the encoder to be  `[None for _ in range(len(self.layers))]` to preserve `head_mask[i]` and keep a consistency across expected arguments.

**Comment by manoja328** [comment]
Yup, that should be sufficient. I encountered this while I was trying to compute attribution of logits with respect the output logits using jacobians. So there first i computed embeddings for the sequence then if I  pass the embeddings to the encoder to get the logits, which is when I encountered this error.
